charts/squid/templates/squid-deployment.yaml

```diff
@@ -0,0 +1,81 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: squid-service
+  labels:
+    app: squid
+spec:
+  ports:
+  - port: 3128
+  selector:
+    app: squid
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: squid-deployment
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: squid
+  template:
+    metadata:
+      labels:
+        app: squid
+    spec:
+      containers:
+      - name: squid
+        image: {{ .Values.squidImage }}
+        ports:
+        - containerPort: 3128
+          name: squid
+          protocol: TCP
+        volumeMounts:
+        - name: squid-config-volume
+          mountPath: /etc/squid/squid.conf
+          subPath: squid.conf
+        # - name: squid-data
+        #   mountPath: /var/spool/squid
+      - name: exporter
+        image: {{ .Values.exporterImage }}
+        imagePullPolicy: IfNotPresent
+        env:
+        - name: SQUID_HOSTNAME
+          value: localhost
+        - name: SQUID_EXPORTER_LISTEN
+          value: ":9301"
+        - name: SQUID_PORT
+          value: "3128"
+        ports:
+        - name: metrics
+          containerPort: 9301
+          protocol: TCP
+      volumes:
+        - name: squid-config-volume
+          configMap:
+            name: squid-config
+            items:
+            - key: squid.conf
+              path: squid.conf
+---
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: squid-loadgen
+spec:
+  schedule: "*/15 * * * *"
```

Comment:
Every 15 minutes seems high. I think this CronJob could be replaced with a sidecar container in the squid pod.